mxcube3/routes/SampleCentring.py

```diff
@@ -511,6 +490,27 @@ def aClick():
     else:
         return Response(status=409)
 
+def waitForCentringFinishes(): 
+    if mxcube.diffractometer.centringStatus["valid"]:
```

Comment:
From what I see, this doesn't wait : it assumes the centring is finished when it gets called.
As you said it is much better to use signals ; we can't rely on the 3 seconds for all diffractometers, this is not reliable.
I propose you to refactor it like this: create a global `CENTRED_EVENT` event (`gevent.event.Event object`),
connect the centring successful signal and centring failed signals to a global function that would just
do `CENTRED_EVENT.set()`.
You would also need to change the `saveCentringWithId` function to return a generator so Flask can handle other requests while waiting for end of centring:

    if len(clicks) == 3:
        CENTRED_EVENT.wait()
        # then put code like current waitForCentringFinishes()

Hopefully there is nothing else to do ; I know bottle much better than Flask, maybe Flask requires something to be able to handle requests while waiting for the centring event but I hope not since we use gevent it should be automatic.